src/code/c_keyframe.c

```diff
@@ -1,85 +1,1329 @@
+            }
+        }
+    }
+
+    if (IS_ZERO(kfSkelAnime->morphFrames)) {
+        // No morph, just play the animation
+        return FrameCtrl_Update(&kfSkelAnime->frameCtrl);
+    } else if (kfSkelAnime->morphFrames > 0.0f) {
+        // Morph to first frame before playing the animation proper
+        Keyframe_FlexMorphInterpolation(kfSkelAnime);
+        kfSkelAnime->morphFrames -= 1.0f;
+        if (kfSkelAnime->morphFrames <= 0.0f) {
+            kfSkelAnime->morphFrames = 0.0f;
+        }
+        return KEYFRAME_NOT_DONE;
+    } else {
+        // Play the animation immediately, morphing as it plays
+        Keyframe_FlexMorphInterpolation(kfSkelAnime);
+        kfSkelAnime->morphFrames += 1.0f;
+        if (kfSkelAnime->morphFrames >= 0.0f) {
+            kfSkelAnime->morphFrames = 0.0f;
+        }
+        return FrameCtrl_Update(&kfSkelAnime->frameCtrl);
+    }
+}
+
+/**
+ * Draws the limb specified by `limbIndex` of type `KeyFrameFlexLimb` belonging to a flex-type keyframe skeleton to the
+ * display buffer specified by the limb's drawFlags.
+ *
+ * @param limbIndex Pointer to the index of the limb to draw
+ * @param overrideKeyframeDraw Callback for before submitting the limb to be drawn. The matrix state will not include
+ *                             the transformation for the current limb.
+ * @param postKeyframeDraw Callback for after submitting the limb to be drawn. The matrix state will include
+ *                         the transformation for the current limb.
+ * @param arg An arbitrary argument to pass to the callbacks.
+ * @param mtxStack Matrix stack for limb transformations. Should have enough room for one matrix per limb.
+ *
+ * @note Original name unknown
+ */
+void Keyframe_DrawFlexLimb(PlayState* play, KFSkelAnimeFlex* kfSkelAnime, s32* limbIndex,
+                           OverrideKeyframeDrawScaled overrideKeyframeDraw, PostKeyframeDrawScaled postKeyframeDraw,
+                           void* arg, Mtx** mtxStack) {
+    KeyFrameFlexLimb* limb = Lib_SegmentedToVirtual(kfSkelAnime->skeleton->limbs);
+    s32 i;
+    Gfx* newDList;
+    Gfx* limbDList;
+    u8 drawFlags;
+    Vec3f scale;
+    Vec3s rot;
+    Vec3f pos;
+    Vec3s* jointData;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    limb += *limbIndex;
+    jointData = &kfSkelAnime->jointTable[*limbIndex * 3];
+
+    scale.x = jointData->x * 0.01f;
+    scale.y = jointData->y * 0.01f;
+    scale.z = jointData->z * 0.01f;
+
+    jointData++;
+
+    rot.x = jointData->x;
+    rot.y = jointData->y;
+    rot.z = jointData->z;
+
+    jointData++;
+
+    pos.x = jointData->x;
+    pos.y = jointData->y;
+    pos.z = jointData->z;
+
+    Matrix_Push();
+
+    newDList = limbDList = limb->dList;
+    drawFlags = limb->drawFlags;
+
+    if (overrideKeyframeDraw == NULL ||
+        (overrideKeyframeDraw != NULL &&
+         overrideKeyframeDraw(play, kfSkelAnime, *limbIndex, &newDList, &drawFlags, arg, &scale, &rot, &pos) != 0)) {
```

Comment:
```suggestion
         overrideKeyframeDraw(play, kfSkelAnime, *limbIndex, &newDList, &drawFlags, arg, &scale, &rot, &pos))) {
```
To match skelanime.c overrideLimbDraw which also just returns a boolean